contractsV5/FeeHandler.sol

```diff
@@ -0,0 +1,189 @@
+        for(uint i = 0; i < eligibleWallets.length; i ++) {
+            // Internal accounting for rebates per reserve wallet (totalRebatesPerRebateWallet)
+            totalRebatesPerRebateWallet[eligibleWallets[i]] = rebateWei * rebatePercentages[i] / 100;
+            // Internal accounting for total rebates (totalRebates)
+            totalRebates += rebateWei * rebatePercentages[i] / 100;
+            // Internal accounting for reward per epoch (totalRewardsPerEpoch)
+            totalRewardsPerEpoch[epoch] += rewardWei;
+            // Internal accounting for total rewards (totalRewards)
+            totalRewards += rewardWei;
+
+            emit AccumulateReserveRebate(eligibleWallets[i], rebateWei * rebatePercentages[i] / 100);
+        }
+        return true;
+    }
+
+    event DistributeRewards(address staker, uint amountWei);
+
+    function claimStakerReward(address staker, uint percentageInPrecision, uint epoch) public onlyDAO returns(uint) {
+        // Amount of reward to be sent to staker
+        uint amount = totalRewardsPerEpoch[epoch] * percentageInPrecision / 10**18;
+
+        // Update total rewards and total rewards per epoch
+        totalRewardsPerEpoch[epoch] -= amount;
+        totalRewards -= amount;
+
+        // send reward to staker
+        (bool success, ) = staker.call.value(amount)("");
+        require(success, "Transfer of rewards to staker failed.");
+
+        emit DistributeRewards(staker, amount);
+
+        return amount;
+
+    }
+
+    event DistributeRebate(address rebateWallet, uint amountWei);
+
+    // Using rebateWallet instead of reserve so I don't have to store KyberNetwork variable.
+    function claimReserveRebate(address rebateWallet) public returns (uint){
+        // Get total amount of rebate accumulated
+        uint amount = totalRebatesPerRebateWallet[rebateWallet] - 1;
+
+        // Update total rebate and rebate per rebate wallet amounts
+        totalRebates -= amount;
+        totalRebatesPerRebateWallet[rebateWallet] = 1; // Do not use 0 to avoid potential issues with div by 0
+
+        // send rebate to rebate wallet
+        (bool success, ) = rebateWallet.call.value(amount)("");
+        require(success, "Transfer of rebates to rebate wallet failed.");
+
+        emit DistributeRebate(rebateWallet, amount);
+
+        return amount;
+    }
+
+    event BurnKNC(address burnAddress, uint amountWei);
+
+    // we will have to limit amounts. per burn.
+    // and create some block delay between burns.
+    function burnKNC() public returns(uint) {
+        // check if current block > last buy block number + num block interval
+        require(
+            block.number < lastBurnBlock + burnBlockInterval,
+            "Unable to burn as burnBlockInterval has not passed since lastBurnBlock"
+        );
+
+        // update last buy block number
+        lastBurnBlock = block.number;
+
+        // Get srcQty to burn, if greater than 10 ETH, burn only 10 ETH per function call.
```

Comment:
don't use number (10) in comment.
use the constant name.

Reply:
noted, updated and will push fix when other comments are fixed